proto/src/determined/experiment/v1/experiment.proto

```diff
@@ -139,3 +139,21 @@ message ValidationHistoryEntry {
   // the validation.
   float searcher_metric = 3;
 }
+
+// File node is one node of file in experiment model definition file tree.
+message FileNode {
+  // Path of file.
+  string path = 1;
+  // Name of file.
+  string name = 2;
+  // Modification time of file.
+  google.protobuf.Timestamp modified_time = 3;
+  // Number of bytes in file content.
+  int32 content_length = 4;
+  // Is this a directory.
+  bool is_dir = 5;
+  // MIME type of file.
+  string content_type = 6;
```

Comment:
how important is it that we deliver the `content_type` in the filetree?  I imagine the answer is "it's pretty important to the webui" but I think it complicates the filecache implementation quite a bit so I want to make sure.